src/load-script.ts

```diff
@@ -80,8 +85,30 @@ export function loadCustomScript(
             url,
             attributes,
             onSuccess: () => resolve(),
-            onError: () =>
-                reject(new Error(`The script "${url}" failed to load.`)),
+            onError: () => {
+                const defaultError = new Error(
+                    `The script "${url}" failed to load.`
+                );
+
+                if (!window.fetch) {
+                    return reject(defaultError);
+                }
+                // NOTE: Attempt to fetch() the error reason from the response body
+                return fetch(url)
+                    .then((response) => {
+                        // NOTE: Only catch errors from 400 status responses
```

Comment:
I updated the above comment to mention validation errors. I think we can remove this one.
```suggestion
```